oshdb/src/main/java/org/heigit/ohsome/oshdb/OSHDBTimeable.java

```diff
@@ -0,0 +1,40 @@
+package org.heigit.ohsome.oshdb;
+
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+public interface OSHDBTimeable {
+  
+  long getEpochSecond();
+  
+  default OSHDBTimestamp getTimestamp() {
+    return new OSHDBTimestamp(getEpochSecond());
+  }
+  
+  default boolean isBefore(OSHDBTimeable other) {
+    return getEpochSecond() < other.getEpochSecond();
+  }
+  
+  static boolean isBefore(OSHDBTimeable a, OSHDBTimeable b) {
+    return a.isBefore(b);
+  }
+  
+  default boolean isAfter(OSHDBTimeable other) {
+    return getEpochSecond() > other.getEpochSecond();
+  }
+  
+  static boolean isAfter(OSHDBTimeable a, OSHDBTimeable b) {
+    return a.isAfter(b);
+  }
+  
+  static int compare(OSHDBTimeable a, OSHDBTimeable b) {
```

Comment:
why not make the class implement `Comparable<OSHDBTimeable>` instead?

Reply:
:arrow_up: this still waits for an answer. also, didn't we say we want to rename this interface (see https://github.com/GIScience/oshdb/pull/369#pullrequestreview-641250682)?

Reply:
OSHDBTimeable now called OSHDBTemporal

>    why not make the class implement Comparable instead?

 you mean the interface OSHDTimeable/OSHDBTemporal should extend Comparable right? This is actually quite problematic, because all implementations of this interface would then inherit the Comparable which is in most case not the desired behavior. The reason for the static compare is to make it more convenient to use it as a Comparator function in e.g. TreeMap/Sets. We can leave it in the interface or remove it.

Reply:
> […] because all implementations of this interface would then inherit the Comparable which is in most case not the desired behavior. The reason for the static compare is to make it more convenient to use it as a Comparator function in e.g. TreeMap/Sets. We can leave it in the interface or remove it.

OK, that makes sense. Let's leave it as is then.